### PR TITLE
enable the diagnostics module

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit enables the diagnostics module and refactors the script `otbr-setup.bash` to put the common build options together.

### CODE REVIEW

File: script/otbr-setup.bash [hunk 1]
```diff
@@ -41,74 +41,61 @@ OT_BR_POSIX_COMMIT_HASH=$5
 OTBR_RCP_BUS=$6
 OTBR_RADIO_URL=$7
 
+OTBR_COMMON_OPTIONS="-DOTBR_DUA_ROUTING=ON -DOT_DUA=ON -DOT_MLR=ON -DOT_DIAGNOSTIC=ON -DOT_PACKAGE_VERSION=${OPENTHREAD_COMMIT_HASH} -DOTBR_PACKAGE_VERSION=${OT_BR_POSIX_COMMIT_HASH} -DOT_POSIX_CONFIG_RCP_BUS=${OTBR_RCP_BUS} -DOTBR_RADIO_URL=${OTBR_RADIO_URL}"
```
Comment (librasungirl) [inline script/otbr-setup.bash:44]: move `-DOTBR_DUA_ROUTING=ON -DOT_DUA=ON -DOT_MLR=ON` to  below`OTBR_THREAD_1_2_OPTIONS`

File: script/otbr-setup.bash [hunk 2]
```diff
@@ -41,74 +41,61 @@ OT_BR_POSIX_COMMIT_HASH=$5
...
+OTBR_THREAD_1_2_OPTIONS="-DOT_THREAD_VERSION=1.2 -DOTBR_DNSSD_DISCOVERY_PROXY=OFF -DOTBR_SRP_ADVERTISING_PROXY=OFF -DOT_TREL=OFF"
+OTBR_THREAD_1_3_OPTIONS="-DOT_THREAD_VERSION=1.3 -DOTBR_DNSSD_DISCOVERY_PROXY=ON  -DOTBR_SRP_ADVERTISING_PROXY=ON  -DOT_TREL=ON -DOT_BORDER_ROUTING=ON -DOT_SRP_CLIENT=ON -DOT_DNS_CLIENT=ON -DOT_FULL_LOGS=ON"
```
Comment (librasungirl) [inline script/otbr-setup.bash:46]: For a 1.3.0 reference, it should have also some 1.2 features e.g. -DOTBR_DUA_ROUTING=ON -DOT_DUA=ON -DOT_MLR=ON above

Comment (librasungirl) [reply script/otbr-setup.bash:46]: Should `-DOT_FULL_LOGS=ON`  be moved to `OTBR_COMMON_OPTIONS`?

**Review by librasungirl — APPROVED** [review]

**Comment by sunytt** [comment]
I wonder if you have uploaded the firmware built from this PR to a RPi to verify it works?

**Comment by zhanglongxia** [comment]
> I wonder if you have uploaded the firmware built from this PR to a RPi to verify it works?

Yes. I generated the RaspberryPi image using the command `REFERENCE_PLATFORM=none REFERENCE_RELEASE_TYPE=1.3 IN_CHINA=1 OTBR_RADIO_URL='spinel+hdlc+uart:///spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=115200' ./script/make-reference-release.bash`. And I verified the RaspberryPi image on the RaspberryPi, the otbr-agent is working and the diag module is enabled.

**Review by sunytt — APPROVED** [review]

File: script/otbr-setup.bash [hunk 3]
```diff
@@ -41,74 +41,61 @@ OT_BR_POSIX_COMMIT_HASH=$5
 OTBR_RCP_BUS=$6
 OTBR_RADIO_URL=$7
 
+OTBR_COMMON_OPTIONS="-DOT_DIAGNOSTIC=ON -DOT_FULL_LOGS=ON -DOT_PACKAGE_VERSION=${OPENTHREAD_COMMIT_HASH} -DOTBR_PACKAGE_VERSION=${OT_BR_POSIX_COMMIT_HASH} -DOT_POSIX_CONFIG_RCP_BUS=${OTBR_RCP_BUS} -DOTBR_RADIO_URL=${OTBR_RADIO_URL}"
```
Comment (superwhd) [inline script/otbr-setup.bash:44]: nit: split option string into lines

Comment (superwhd) [inline script/otbr-setup.bash:44]: I'm a little concerned that there may be duplicate/conflicting option entries in the future. Shall we add a check?

File: script/otbr-setup.bash [hunk 4]
```diff
@@ -41,74 +41,87 @@ OT_BR_POSIX_COMMIT_HASH=$5
...
+    "-DOT_MLR=ON"
+    "-DOTBR_DNSSD_DISCOVERY_PROXY=OFF"
+    "-DOTBR_SRP_ADVERTISING_PROXY=OFF"
+    "-DOT_TREL=OFF"
+)
+
+readonly OTBR_THREAD_1_3_OPTIONS=(
+    "-DOT_THREAD_VERSION=1.3"
+    "-DOTBR_DUA_ROUTING=ON"
+    "-DOT_DUA=ON -DOT_MLR=ON"
```
Comment (superwhd) [inline script/otbr-setup.bash:66]: nit: split these two items

**Review by superwhd — APPROVED** [review]
LGTM overall

**Review by superwhd — APPROVED** [review]
LGTM overall

**Pull request merged**